sql/moz-fx-data-shared-prod/stripe/itemized_payout_reconciliation/view.sql

```diff
@@ -0,0 +1,87 @@
+CREATE OR REPLACE VIEW
+  `moz-fx-data-shared-prod.stripe.itemized_payout_reconciliation`
+AS
+WITH customers AS (
+  SELECT
+    id,
+    address.country,
+    NULLIF(address.postal_code, "") AS postal_code,
+    NULLIF(address.state, "") AS state,
+  FROM
+    `moz-fx-data-shared-prod`.stripe_external.customers_v1
+  WHERE
+    address.country IN ("US", "CA")
+),
+postal_code_to_state AS (
+  SELECT
+    country,
+    postal_code,
+    IF(
+      COUNT(DISTINCT state) > 1,
+      ERROR("postal code resolved to mulitple states: " || TO_JSON_STRING(postal_code)),
+      ANY_VALUE(state)
+    ) AS state,
+  FROM
+    customers
+  WHERE
+    postal_code IS NOT NULL
+  GROUP BY
+    country,
+    postal_code
+),
+charges AS (
+  SELECT
+    id,
+    payment_method_details.card.country,
+    NULLIF(billing_details.address.postal_code, "") AS postal_code,
+    state,
+  FROM
+    `moz-fx-data-shared-prod`.stripe_external.charges_v1
+  JOIN
+    postal_code_to_state
+  ON
+    payment_method_details.card.country = postal_code_to_state.country
+    AND billing_details.address.postal_code = postal_code_to_state.postal_code
+  WHERE
+    payment_method_details.card.country IN ("US", "CA")
+),
+enriched AS (
+  SELECT
+    report.*,
+    charges.state,
+    charges.postal_code,
+  FROM
+    `moz-fx-data-shared-prod`.stripe_external.itemized_payout_reconciliation_v5 AS report
+  LEFT JOIN
+    charges
+  ON
+    charge_id = charges.id
+    AND card_country = charges.country
+)
+SELECT
+  * EXCEPT (card_country, state),
+  CASE
+  -- american samoa
+  WHEN
+    card_country = "US"
+    AND REGEXP_CONTAINS(postal_code, "^96799(-?[0-9]{4})?$")
+  THEN
+    STRUCT("AS" AS card_country, NULL AS state)
+  -- puerto rico
+  WHEN
+    card_country = "US"
+    AND REGEXP_CONTAINS(postal_code, "^00[679][0-9]{2}(-?[0-9]{4})?$")
+  THEN
+    STRUCT("PR" AS card_country, NULL AS state)
+  -- virgin islands
+  WHEN
+    card_country = "US"
+    AND REGEXP_CONTAINS(postal_code, "^008[0-9]{2}(-?[0-9]{4})?$")
+  THEN
+    STRUCT("VI" AS card_country, NULL AS state)
+  ELSE
+    STRUCT(card_country, state)
+  END
```

Comment:
see https://mozilla-hub.atlassian.net/browse/DVPN-106?focusedCommentId=515873 for why this is needed

Reply:
indeed, postal code needs to be trimmed